### PR TITLE
'ColumnInfo is locked' related test failure fix (for TargetStudyTest.runUITests).

### PR DESCRIPTION
**Pull request by labkey-bpatel**

### CODE REVIEW

File: api/src/org/labkey/api/study/actions/StudyPickerColumn.java [hunk 1]
```diff
@@ -53,7 +53,12 @@ public StudyPickerColumn(ColumnInfo col, String inputName)
     {
         super(AbstractAssayProvider.TARGET_STUDY_PROPERTY_CAPTION, inputName);
         _colInfo = col;
-        ((BaseColumnInfo)_colInfo).setInputType("select");
+        BaseColumnInfo baseColumnInfo = (BaseColumnInfo) _colInfo;
+        if (baseColumnInfo.isLocked())
+        {
+            baseColumnInfo.setLocked(false);
+        }
```
Comment (labkey-bpatel) [inline api/src/org/labkey/api/study/actions/StudyPickerColumn.java]: @labkey-matthewb 
Main change I made to this method was setting lock to false if it was locked to fix a test failure:
Does this look like a reasonable change?

File: api/src/org/labkey/api/study/actions/StudyPickerColumn.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ public StudyPickerColumn(ColumnInfo col, String inputName)
         {
             baseColumnInfo.setLocked(false);
         }
-        baseColumnInfo.setInputType("select");
+        baseColumnInfo.setInputType(col.getInputType());
```
Comment (labkey-bpatel) [inline api/src/org/labkey/api/study/actions/StudyPickerColumn.java]: @labkey-matthewb 
As per your suggestion, for the previously existing _setInputType("select")_ is now changed to _setInputType(col.getInputType())_

Comment (labkey-matthewb) [reply api/src/org/labkey/api/study/actions/StudyPickerColumn.java]: I think you want

_inputType = "select"

Comment (labkey-bpatel) [reply api/src/org/labkey/api/study/actions/StudyPickerColumn.java]: _inputType variable is not accessible in StudyPickerColumn for me to set it unless I should be creating one? Sorry if im missing something here.

File: api/src/org/labkey/api/study/actions/StudyPickerColumn.java [hunk 3]
```diff
@@ -58,7 +58,6 @@ public StudyPickerColumn(ColumnInfo col, String inputName)
         {
             baseColumnInfo.setLocked(false);
         }
-        baseColumnInfo.setInputType(col.getInputType());
```
Comment (labkey-bpatel) [inline api/src/org/labkey/api/study/actions/StudyPickerColumn.java]: @labkey-matthewb 
As per our email conversation, this line of code is now removed.
Thanks for your insight!

**Pull request merged**